sunpy/map/mapbase.py

```diff
@@ -476,12 +476,23 @@ def wcs(self):
         if w2.wcs.ctype[1].lower() in ("solar-y", "solar_y"):
             w2.wcs.ctype[1] = 'HPLT-TAN'
 
-        # GenericMap.coordinate_frame is implemented using this method, so we
-        # need to do this only based on .meta.
-        ctypes = {c[:4] for c in w2.wcs.ctype}
-        # Check that the ctypes contains one of these three pairs of axes.
-        if ({'HPLN', 'HPLT'} <= ctypes or {'SOLX', 'SOLY'} <= ctypes or {'CRLN', 'CRLT'} <= ctypes):
-            w2.heliographic_observer = self.observer_coordinate
+        # Set observer coordinate information
+        #
+        # Clear all the aux information that was set earlier. This is to avoid
+        # issues with maps that store multiple observer coordinate keywords.
+        # Note that we have to create a new WCS as it's not possible to modify
+        # wcs.wcs.aux in place.
+        header = w2.to_header()
+        for kw in ['crln_obs', 'dsun_obs', 'hgln_obs', 'hglt_obs']:
+            header.pop(kw, None)
```

Comment:
is this case sensitive? I don't think so right?

Reply:
Looking in the code for `astropy.io.fits.Header` it seems like it is, though it's not clear from the docs. Opened https://github.com/astropy/astropy/issues/11031 but seems safe to take this approach for now.

Reply:
Confirmed with tests in https://github.com/astropy/astropy/pull/11039 that this is fine.